library/Odesk/Phystrix/AbstractCommand.php

```diff
@@ -219,33 +219,32 @@ public function execute()
         if ($this->isRequestCacheEnabled()) {
             $fromCache = $this->requestCache->get($this->getCommandKey(), $this->getCacheKey());
             if ($fromCache !== null) {
-                $this->executionEvents[] = self::EVENT_RESPONSE_FROM_CACHE;
                 $metrics->markResponseFromCache();
+                $this->handleEvent(self::EVENT_RESPONSE_FROM_CACHE);
                 return $fromCache;
             }
         }
         $circuitBreaker = $this->getCircuitBreaker();
         if (!$circuitBreaker->allowRequest()) {
-            $this->executionEvents[] = self::EVENT_SHORT_CIRCUITED;
             $metrics->markShortCircuited();
+            $this->handleEvent(self::EVENT_SHORT_CIRCUITED);
             return $this->getFallbackOrThrowException();
         }
         $this->invocationStartTime = $this->getTimeInMilliseconds();
         try {
             $result = $this->run();
             $this->recordExecutionTime();
-            $this->executionEvents[] = self::EVENT_SUCCESS;
             $metrics->markSuccess();
             $circuitBreaker->markSuccess();
+            $this->handleEvent(self::EVENT_SUCCESS);
         } catch (BadRequestException $exception) {
             // Treated differently and allowed to propagate without any stats tracking or fallback logic
             $this->recordExecutionTime();
             throw $exception;
         } catch (Exception $exception) {
             $this->recordExecutionTime();
-            $this->executionException = $exception;
-            $this->executionEvents[] = self::EVENT_FAILURE;
             $metrics->markFailure();
+            $this->handleEvent(self::EVENT_FAILURE, $exception);
```

Comment:
> as phystrix does not (yet) have time-limiting or segfault-bypassing powers
You should configure a reasonable timeout for the underlying remote call (http connection timeout, db timeout etc) so that an exception is thrown when it happens.